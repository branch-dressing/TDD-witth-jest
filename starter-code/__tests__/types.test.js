const {
  isNumber,
  isString,
  isBoolean,
  isArray,
  castToNumber,
  getCaster
} = require('../lib/types.js');

describe('validator module', () => {
  describe('basic validation', () => {
    it('properly tells if a value is a numbers', () => {
      expect(isNumber(3)).toBeTruthy();
      expect(isNumber('hi')).toBeFalsy();
      expect(isNumber([])).toBeFalsy();
      expect(isNumber({})).toBeFalsy();
      expect(isNumber(() => {})).toBeFalsy();
      expect(isNumber(true)).toBeFalsy();
    });

    it('tells if a value is a string', () => {
      expect(isString('hello')).toBeTruthy();
      expect(isString('1')).toBeTruthy();
      expect(isString(1)).toBeFalsy();
      expect(isString([])).toBeFalsy();
      expect(isString({})).toBeFalsy();
      expect(isString(() => {})).toBeFalsy();
    });

    it('tells if a value is a boolean', () => {
      expect(isBoolean(true)).toBeTruthy();
      expect(isBoolean(false)).toBeTruthy();
      expect(isBoolean('hello')).toBeFalsy();
      expect(isBoolean('1')).toBeFalsy();
      expect(isBoolean(1)).toBeFalsy();
      expect(isBoolean([])).toBeFalsy();
      expect(isBoolean({})).toBeFalsy();
      expect(isBoolean(() => {})).toBeFalsy();
    });

    it('tells if a value is an array', () => {
      expect(isArray([])).toBeTruthy();
      expect(isArray([1, 2, 3])).toBeTruthy();
      expect(isArray(['This', 'is', 'an', 'Array!'])).toBeTruthy();
      expect(isArray(true)).toBeFalsy();
      expect(isArray(false)).toBeFalsy();
      expect(isArray('hello')).toBeFalsy();
      expect(isArray('1')).toBeFalsy();
      expect(isArray(1)).toBeFalsy();
      expect(isArray({})).toBeFalsy();
      expect(isArray(() => {})).toBeFalsy();
    })
  });

  describe('casters', () => {
    it('can cast values to a number', () => {
      expect(castToNumber(3)).toEqual(3);
      expect(castToNumber('3')).toEqual(3);
      expect(castToNumber(true)).toEqual(1);
      expect(castToNumber(false)).toEqual(0);
    });

    it('throws if value is not castable to number', () => {
      expect(() => castToNumber('hi')).toThrowErrorMatchingSnapshot();
      expect(() => castToNumber({})).toThrowErrorMatchingSnapshot();
    });
  });

  it('can get the right caster', () => {
    expect(getCaster(Number)).toEqual(castToNumber);
    expect(getCaster(Promise)).toBeNull();
  });
});
