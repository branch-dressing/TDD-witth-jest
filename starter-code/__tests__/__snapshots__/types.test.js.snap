// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validator module casters throws if value is not castable to number 1`] = `"Cannot cast >>hi<< to Number"`;

exports[`validator module casters throws if value is not castable to number 2`] = `"Cannot cast >>[object Object]<< to Number"`;

exports[`validator module throws if value is not castable to a string 1`] = `"Cannot cast >>null<< to String"`;

exports[`validator module throws if value is not castable to a string 2`] = `"Cannot cast >>undefined<< to String"`;

exports[`validator module throws if value is not castable to an array 1`] = `"Cannot cast >>null<< to Array"`;

exports[`validator module throws if value is not castable to an array 2`] = `"Cannot cast >>undefined<< to Array"`;

exports[`validator module throws if value is not castable to an array 3`] = `"Cannot cast >>() => {}<< to Array"`;
